src/megaclient.cpp

```diff
@@ -7852,6 +7858,36 @@ error MegaClient::rename(Node* n, Node* p, syncdel_t syncdel, handle prevparent,
     return API_OK;
 }
 
+void MegaClient::removeOutSharesFromSubtree(Node* n)
+{
+    if (n->pendingshares)
+    {
+        for (auto& it : *n->pendingshares)
+        {
+            if (it.second->pcr)
+            {
+                setshare(n, it.second->pcr->targetemail.c_str(), ACCESS_UNKNOWN, nullptr);
+            }
+        }
+    }
+
+    if (n->outshares)
+    {
+        for (auto& it : *n->outshares)
+        {
+            if (it.second->user)
+            {
+                setshare(n, it.second->user->email.c_str(), ACCESS_UNKNOWN, nullptr);
+            }
```

Comment:
```suggestion
            }
            else    // folder links are a shared folder without user
            {
                setshare(n, nullptr, ACCESS_UNKNOWN, nullptr);
            }
```

Reply:
Excellent point.   As lu1 was not available for me either, I've made a new PR with these changes, against `develop`.  And I've incorporated this change too.   All 4 cases that you've listed above are working.   thanks!